resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -216,7 +228,18 @@ private[spark] class Client(
           .endEnv()
         .endContainer()
       .endSpec()
-      .build()
+    val resolvedDriverPod = if (!isPython) {
```

Comment:
Can we try to boil all of the logic guarded by the `isPython` flag into one place? Perhaps a separate class altogether and we create an Option of that class?